aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java

```diff
@@ -403,10 +396,10 @@ public static ModifyDbInstanceRequest modifyDbInstanceRequest(
 
         if (BooleanUtils.isTrue(isRollback)) {
             builder.allocatedStorage(
-                    canUpdateAllocatedStorage(previousModel.getAllocatedStorage(), desiredModel.getAllocatedStorage()) ? getAllocatedStorage(desiredModel) : getAllocatedStorage(previousModel)
```

Comment:
Is it intentional that `modifyDbInstanceRequestV12` and `modifyDbInstanceRequest` use different conditions?